[.github/workflows] Added pigweed-app related builds to workflows.

kkasperczyk-no opened the pull request:
#### Problem
There is no build validation in CI for pigweed-app, as this app related builds are not ran in the github workflows.

 #### Summary of Changes
* Modified nrfconnect_example.sh to take board name as an mandatory argument and also allow passing list of additional arguments (e.g. attaching overlays).
* Added pigweed-app build and lighting-app with rpc.overlay build to the nrfconnect github workflows.
* Renamed esp_echo_app.sh to the esp_example.sh and modified it to take application name as an argument.
* Added pigweed-app build to the esp32 github workflow.
* Renamed Echo App build step to the All Clusters App in the esp32 and qemu github workflows, as in fact that app is being build.

 Fixes #4562

szatmz commented on scripts/examples/nrfconnect_example.sh:
```diff
@@ -16,12 +16,15 @@
-# Run bootstrap to set up e.g. Pigweed correctly
+# Run bootstrap and activate to set up e.g. Pigweed correctly
 source "$(dirname "$0")/../../scripts/bootstrap.sh"
+source "$(dirname "$0")/../../scripts/activate.sh"
```
isn't running bootstrap.sh enough?

kkasperczyk-no replied:
I think not for pigweed-app, as activate.sh sets proper python version. Without running activate.sh python version is set on the wrong version (I guess it is 2.7 ?) and when compiling pigweed syntax errors appear.

szatmz replied:
Well, that's weird given that activate.sh is a symlink to bootstrap.sh:

```$ ll ./scripts/activate.sh 
lrwxrwxrwx 1 szatmz primarygroup 12 Nov  2 17:48 ./scripts/activate.sh -> bootstrap.sh
```

woody-apple replied:
@kkasperczyk-no can you take a peek at this? Sounds like something isn't working as expected..

szatmz reviewed (approved):
Awesome, thank you!

LuDuda commented on .github/workflows/examples-nrfconnect.yaml:
```diff
@@ -45,12 +45,14 @@ jobs:
+            - name: Build example nRF Connect SDK Lock App on nRF52840 DK
+              run: scripts/examples/nrfconnect_example.sh lock-app nrf52840dk_nrf52840
+            - name: Build example nRF Connect SDK Lighting App on nRF52840 DK
+              run: scripts/examples/nrfconnect_example.sh lighting-app nrf52840dk_nrf52840 -DOVERLAY_CONFIG=rpc.overlay
```
I would say, we should build the Lighting app without overlay as well as it's the default configuration.

kkasperczyk-no replied:
Ok, I misunderstood that we want to run builds with overlay by default, fixed.

LuDuda commented on .vscode/tasks.json:
```diff
@@ -194,9 +194,9 @@
+            "label": "Build nRF Connect Lighting Example (nRF52840)",
             "type": "shell",
-            "command": "source scripts/activate.sh && scripts/examples/nrfconnect_example.sh lighting-app",
+            "command": "source scripts/activate.sh && scripts/examples/nrfconnect_example.sh lighting-app nrf52840dk_nrf52840 -DOVERLAY_CONFIG=rpc.overlay",
```
I would not add the overlay here, but keep the default configuration.

LuDuda commented on scripts/examples/nrfconnect_example.sh:
```diff
@@ -30,8 +33,13 @@ if [[ ! -f "$APP/nrfconnect/CMakeLists.txt" ]]; then
 env
 
-west build -b nrf52840dk_nrf52840 -d "$APP/nrfconnect/build" "$APP/nrfconnect"
+west build -b $BOARD -d "$APP/nrfconnect/build" "$APP/nrfconnect" -- "$@"
```
This should fix the restyled error.

```suggestion
west build -b "$BOARD" -d "$APP/nrfconnect/build" "$APP/nrfconnect" -- "$@"
```

kkasperczyk-no replied:
Great, that did the job, thanks!

LuDuda reviewed (changes requested):
Thanks :+1: few comments below.

woody-apple commented:
@kkasperczyk-no Note: build fails :(

kkasperczyk-no commented:
> @kkasperczyk-no Note: build fails :(

Yes I know and it also fails for me locally, but I'm not sure if it is related to my change or the pigweed-app for ESP32 build doesn't work and now I added it to CI, so I triggered fail. I contacted with Zoltan, so I have a hope to get some help with that.

szatmz commented:
https://github.com/project-chip/connectedhomeip/pull/4563/files#diff-470dd5fd8deeed802cd57c2e1d4b0aa41289828b479df1bdfd1cde5cd75c4a48R41 'all-clusters-app' needs replaced with '"$root"', no?

szatmz commented:
Regarding the ESP32 build breakage:

The problem is that esp_example.sh uses two sdkconfig defaults to seed the sdkconfig for the build: sdkconfig_devkit.defaults and sdkconfig_m5stack.defaults. The directory examples/pigweed-app/esp32 has neither.
The esp_example.sh script should be refactored. It should look for all the sdkconfig*.defaults files and run the build for all of those that actually exist in a particular example app's directory.
Please add this refactoring to your PR. Many thanks in advance!

kkasperczyk-no commented:
> Regarding the ESP32 build breakage:
> 
> The problem is that esp_example.sh uses two sdkconfig defaults to seed the sdkconfig for the build: sdkconfig_devkit.defaults and sdkconfig_m5stack.defaults. The directory examples/pigweed-app/esp32 has neither.
> The esp_example.sh script should be refactored. It should look for all the sdkconfig*.defaults files and run the build for all of those that actually exist in a particular example app's directory.
> Please add this refactoring to your PR. Many thanks in advance!

@szatmz now it's working, but I had to change several things, so please verify if you are ok with them:
- Added for loop building all sdkconfig*.defaults present in the given app directory
- Added running activate.sh before building to prevent pigweed-app from failing because of the incorrect python version
- Changed target name for the esp32 pigweed-app from pw-rpc-app to chip-pigweed-app to be consistent with chip-all-clusters-app.

szatmz commented:
This looks good to me now, and apparently the build checks are passing too. Thank you Kamil for your work on this. The need to call 'activate.sh' is peculiar, I will ask Pigweed devs about it. Nevertheless that shouldn't hold this PR back from merging.
